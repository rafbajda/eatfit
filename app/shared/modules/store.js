import { createStore, applyMiddleware, compose, combineReducers } from 'redux';
import thunk from 'redux-thunk';
import { getFirestore, reduxFirestore, firestoreReducer } from 'redux-firestore';
import { getFirebase, reactReduxFirebase, firebaseReducer } from 'react-redux-firebase';
import { composeWithDevTools } from 'redux-devtools-extension';
import {
    createReactNavigationReduxMiddleware,
    createNavigationReducer,
} from 'react-navigation-redux-helpers';
import initialState from '../constants/State';
import loginReducer from '../../screens/login/state/reducer';
import firebaseConfig from './firebase';
import configMiddleware from '../../screens/login/state/middleware';
import RootNavigator from '../../navigation/RootNavigator';

let store = null; // eslint-disable-line
const navReducer = createNavigationReducer(RootNavigator);
const navMiddleware = createReactNavigationReduxMiddleware(state => state.nav);

const combinedReducer = combineReducers({
    config: loginReducer,
    nav: navReducer,
    firestore: firestoreReducer,
    firebase: firebaseReducer,
});

if (__DEV__) {
    // eslint-disable-line
  const devToolsEnhancer = require('remote-redux-devtools'); // eslint-disable-line
    store = createStore(
        combinedReducer,
        initialState,
        composeWithDevTools(
            applyMiddleware(
                navMiddleware,
                configMiddleware,
                thunk.withExtraArgument({ getFirebase, getFirestore })
            ),
            reactReduxFirebase(firebaseConfig),
            reduxFirestore(firebaseConfig)
        )
    );
} else {
    store = createStore(
        combinedReducer,
        initialState,
        compose(
            applyMiddleware(
                navMiddleware,
                configMiddleware,
                thunk.withExtraArgument({ getFirebase, getFirestore })
            ),
            reduxFirestore(firebaseConfig),
            reactReduxFirebase(firebaseConfig)
        )
    );
}

export default store;
