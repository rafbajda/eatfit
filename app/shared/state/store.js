/* global __DEV__ */
import { createStore, applyMiddleware, compose, combineReducers } from 'redux';
import thunk from 'redux-thunk';
import { getFirestore, reduxFirestore, firestoreReducer } from 'redux-firestore';
import { getFirebase, reactReduxFirebase, firebaseReducer } from 'react-redux-firebase';
import { composeWithDevTools } from 'redux-devtools-extension';
import {
    createReactNavigationReduxMiddleware,
    createNavigationReducer,
} from 'react-navigation-redux-helpers';
import globalReducers from './reducer';
import firebaseConfig from '../modules/firebase';
import RootNavigator from '../../navigation/RootNavigator';
import initialState from '../constants/State';
import globalMiddlewares from './middleware';
import authReducer from '../../screens/login/state/reducer';
import signUpMiddleware from '../../screens/signUp/state/middleware';

let store = null; // eslint-disable-line
const navReducer = createNavigationReducer(RootNavigator);
const navMiddleware = createReactNavigationReduxMiddleware(state => state.nav);

const combinedReducer = combineReducers({
    auth: authReducer,
    config: globalReducers.configReducer,
    nav: navReducer,
    firestore: firestoreReducer,
    firebase: firebaseReducer,
});

if (__DEV__) {
    // eslint-disable-line
  const devToolsEnhancer = require('remote-redux-devtools'); // eslint-disable-line
    store = createStore(
        combinedReducer,
        initialState,
        composeWithDevTools(
            applyMiddleware(
                navMiddleware,
                globalMiddlewares.configMiddleware,
                signUpMiddleware,
                thunk.withExtraArgument({ getFirebase, getFirestore })
            ),
            reactReduxFirebase(firebaseConfig),
            reduxFirestore(firebaseConfig)
        )
    );
} else {
    store = createStore(
        combinedReducer,
        initialState,
        compose(
            applyMiddleware(
                navMiddleware,
                globalMiddlewares.configMiddleware,
                signUpMiddleware,
                thunk.withExtraArgument({ getFirebase, getFirestore })
            ),
            reduxFirestore(firebaseConfig),
            reactReduxFirebase(firebaseConfig)
        )
    );
}

export default store;
