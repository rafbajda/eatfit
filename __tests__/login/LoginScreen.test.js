import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import LoginScreen from "../../app/screens/login";
import store from "../../app/shared/state/store";
import {Provider} from "react-redux";

describe('LoginScreen snapshot', () => {
    jest.useFakeTimers();
    beforeEach(() => {
        NavigationTestUtils.resetInternalState();
    });

    it('renders component', async () => {
        const tree = renderer.create(<Provider store={store}><LoginScreen/></Provider>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
