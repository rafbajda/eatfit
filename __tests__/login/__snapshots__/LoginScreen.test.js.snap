// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen snapshot renders component 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  className="sc-bdVaJa bDWFJH"
  contentContainerStyle={
    Array [
      Object {
        "padding": undefined,
      },
      Object {
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutomaticScroll={true}
  enableOnAndroid={false}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  getScrollResponder={[Function]}
  handleOnScroll={[Function]}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  keyboardShouldPersistTaps="handled"
  keyboardSpace={0}
  onScroll={[Function]}
  resetKeyboardSpace={[Function]}
  resetScrollToCoords={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scrollEventThrottle={1}
  scrollForExtraHeightOnAndroid={[Function]}
  scrollIntoView={[Function]}
  scrollToEnd={[Function]}
  scrollToFocusedInput={[Function]}
  scrollToPosition={[Function]}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
  update={[Function]}
  viewIsInsideTabBar={false}
>
  <View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <View
        animating={true}
        animationDuration={600}
        animationEasing={[Function]}
        color="#1db954"
        interaction={true}
        renderComponent={[Function]}
        style={
          Object {
            "height": 200,
            "width": 250,
          }
        }
      >
        <View
          style={
            Object {
              "height": 100,
              "left": 0,
              "position": "absolute",
              "top": 50,
              "transform": Array [
                Object {
                  "rotate": "0deg",
                },
              ],
              "width": 100,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "height": 50,
                "overflow": "hidden",
                "width": 100,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#1db954",
                  "borderRadius": 50,
                  "height": 100,
                  "width": 100,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "height": 100,
              "left": 0,
              "position": "absolute",
              "top": 50,
              "transform": Array [
                Object {
                  "rotate": "0deg",
                },
              ],
              "width": 100,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "height": 50,
                "overflow": "hidden",
                "top": 50,
                "width": 100,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#1db954",
                  "borderRadius": 50,
                  "height": 100,
                  "top": -50,
                  "width": 100,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#1db954",
              "borderRadius": 12.5,
              "height": 25,
              "left": 112.5,
              "position": "absolute",
              "top": 87.5,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "scale": 1,
                },
              ],
              "width": 25,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#1db954",
              "borderRadius": 12.5,
              "height": 25,
              "left": 162.5,
              "position": "absolute",
              "top": 87.5,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "scale": 1,
                },
              ],
              "width": 25,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#1db954",
              "borderRadius": 12.5,
              "height": 25,
              "left": 212.5,
              "opacity": 0,
              "position": "absolute",
              "top": 87.5,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "scale": 0,
                },
              ],
              "width": 25,
            }
          }
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;
